### PR TITLE
nova-editor: Update types for Nova 2

### PR DESCRIPTION
**Pull request by apexskier**
The biggest feature here is new dynamic Tasks and some language client improvements

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://library.panic.com/releasenotes/nova/#2
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- ~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.~

### CODE REVIEW

**Comment by logancollins** [comment]
This all looks good from the Nova 2 side of things!

**Review by logancollins — APPROVED** [review]
Approved from my perspective!

**Comment by apexskier** [comment]
Ready to merge

File: types/nova-editor-node/index.d.ts [hunk 1]
```diff
@@ -19,25 +19,27 @@ type WritableStream<T = any> = unknown;
...
 interface AssistantsRegistry {
-    registerColorAssistant(selector: AssistantsRegistrySelector, object: ColorAssistant): Disposable;
     registerCompletionAssistant(selector: AssistantsRegistrySelector, object: CompletionAssistant): Disposable;
     registerIssueAssistant(
         selector: AssistantsRegistrySelector,
         object: IssueAssistant,
-        options?: { event: "onChange" | "onSave" }
+        options?: { event: 'onChange' | 'onSave' },
     ): Disposable;
+    registerTaskAssistant(object: TaskAssistant, options?: { identifer: string; name: string }): Disposable;
```
Comment (jaydenseric) [inline types/nova-editor-node/index.d.ts:28]: This should have been `identifier`, not `identifer`?

Comment (apexskier) [reply types/nova-editor-node/index.d.ts:28]: yeah, looks like a typo

**Pull request merged**